ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
-        "each layer. It indicates the norm of the output and can be null, batch or layer.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["norm"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["tiny", "small", "base", "large"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVConvNeXtEncoder"]["model_variant"],
     )
 
-    norm_params: Optional[Dict[str, Any]] = schema_utils.Dict(
-        default=None,
-        description="parameters used if norm is either batch or layer. For information on parameters used with batch "
-        "see Torch's documentation on batch normalization or for layer see Torch's documentation on layer "
-        "normalization.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["norm_params"],
+
+@register_encoder_config("densenet_torch", IMAGE)
+@dataclass
+class TVDenseNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["densenet_torch"],
+        default="densenet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    num_fc_layers: Optional[Optional[int]] = schema_utils.PositiveInteger(
-        default=1,
-        description="The number of stacked fully connected layers.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["num_fc_layers"],
+    model_variant: Optional[int] = schema_utils.IntegerOptions(
+        [121, 161, 169, 201],
+        default=121,
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVDenseNetEncoder"]["model_variant"],
     )
 
-    fc_layers: Optional[Optional[List[Dict]]] = schema_utils.DictList(
-        default=None,
-        description="A list of dictionaries containing the parameters of all the fully connected layers. The length "
-        "of the list determines the number of stacked fully connected layers and the content of each "
-        "dictionary determines the parameters for a specific layer. The available parameters for each "
-        "layer are: activation, dropout, norm, norm_params, output_size, use_bias, bias_initializer and "
-        "weights_initializer. If any of those values is missing from the dictionary, the default one "
-        "specified as a parameter of the encoder will be used instead. ",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["fc_layers"],
+
+@register_encoder_config("efficientnet_torch", IMAGE)
+@dataclass
+class TVEfficientNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["efficientnet_torch"],
+        default="efficientnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional